### ISSUE TITLE
Expo App Broke but my Application is working

### ISSUE THREAD

**Issue by Leny73** [expo/expo#3206]
# READ THIS FIRST, PLEASE!

Hello! Thanks for reporting an issue. Please make sure you're posting this in the right place:

1. If this is an issue with Expo CLI, our command line tool, please post it in the
  https://github.com/expo/expo-cli repo instead.
2. If this is a feature request, please vote or post it at https://expo.canny.io instead.
3. If you are unable to create a reproducible Snack example that demonstrates the bug/issue,
  please first post in our forums at https://forums.expo.io/ before opening an issue here!
  (Unless the bug/issue pertains to a standalone, non-ejected build)

Thanks for helping us make Expo better!

### Environment

 Expo CLI 2.6.14 environment info:
    System:
      OS: Windows 7
    Binaries:
      Yarn: 1.9.4 - ~\AppData\Roaming\npm\yarn.CMD
      npm: 6.4.1 - C:\Program Files\nodejs\npm.CMD

iOS, Android, Standalone

### Steps to Reproduce

(Write your steps here:)

1.go to https://expo.io/@leny73/culture-map-plovdiv
2.open app using expo
3.reload

### Expected Behavior
Opening my application without errors

(Write what you thought would happen.)

### Actual Behavior
  Opening my application from https://expo.io/@leny73/culture-map-plovdiv It opens and everything is ok after 3-4-5 second It gives me a blue screen on which there is a message Something went wrong. Sorry about that . You can go back to Expo home or try to reload the Experience. When you reload the application is running pefectly.When I try to go to home it doesn't let me navigate to the Home screen of the appliaction. This happened after a production build. The Expo app was running perfectly but after the production build when I start Expo it gives my another error which is [Uncaught Error: Module AppRegistry is not a registered callable module(calling runApplication) value@cached-bundle-kernel471464494-UNVERSIONED:28] and I can't navigate in the application.
The error my colleague gets is Uncaught Error: undefined is not an object (evaluating 'c.default.Event_Name)
<unknown>@cached-bundle-kernel471464494-UNVERSIONED:391 Fatal Error
The errors I recieve are 
[Uncaught Error: Module AppRegistry is not a registered callable module(calling runApplication) value@cached-bundle-kernel471464494-UNVERSIONED:28] 
[Uncaught Error: undefined is not an object (evaluating 'c.default.Event_Name)
<unknown>@cached-bundle-kernel471464494-UNVERSIONED:391 Fatal Error]

(Write what happened. Add screenshots!)

### Reproducible Demo

(Paste the link to an example project and exact instructions to reproduce the issue.)

**Comment by maderesponsively** [comment]
I re-installed the EXPO app today (9th Jan update) and I too receive the blue screen with "Something went wrong. Sorry about that. You can go back to Expo home or try to reload the Experience."

I have tried uninstalling and reinstalling but with no resolve. Any ideas?

**Comment by Hotsuyuki** [comment]
@Leny73 
I had the exactly same error message when I tried to open Expo Client App on android (on the other hand, Expo Client App on iOS is no problem).

@maderesponsively 
I think you are using Expo Client App on android, right? I have the same version which is 2.10.3 released on Jan 9th 2019, and got same error message.


Below are my status.

## OS: 
Android 7.0

## Expo Client App version: 
2.10.3 released on Jan 9th 2019

## Actual Behavior:
When opening Expo Client App on android, before landing to the sign in page, I got a blue screen with the message like 
"Something went wrong"
"Sorry about that. You can go back to Expo home or try to reload the Experience."

![screenshot_20190113-000828](https://user-images.githubusercontent.com/26198619/51074919-51040900-16c8-11e9-9774-0edb66d2d349.jpg)

Then I tap "View error log", I got 
"Uncaught Error: Module AppRegistry is not a registered callable module(calling runApplication) value@cached-bundle-kernel471464494-UNVERSIONED:28"
"Uncaught Error: undefined is not an object (evaluating 'c.default.Event_Name)
@cached-bundle-kernel471464494-UNVERSIONED:391 Fatal Error"

![screenshot_20190113-000832](https://user-images.githubusercontent.com/26198619/51074927-5feabb80-16c8-11e9-8db8-1515ae708da1.jpg)

FYI, the very first launching Expo Client App right after downloading from the Google play would be succeeded, but second time or later it wouldn't open (getting the above messages).
Expo Client App on iOS (ver. 2.10.0) is no problem, everything works fine.

**Comment by maderesponsively** [comment]
Update: 
![screenshot 2019-01-12 at 16 08 31](https://user-images.githubusercontent.com/6838179/51075532-68b99e00-1684-11e9-8df8-272512f335d3.png)

**Comment by maderesponsively** [comment]
@sjchmiela tried doing as you suggested, it's still not working 😔

**Comment by sjchmiela** [comment]
🤔 Maybe try that a couple of times? Maybe trying reloading the app with that _reload_ button? Sorry for the inconvenience, I'm not sure there's some deterministic way of requesting an OTA update to Home…

**Comment by maderesponsively** [comment]
@sjchmiela the restarts and kill (10s of times) didn't do it but the delete app and reinstall did! It's working now 😅

**Comment by sjchmiela** [comment]
Phew! 😅

**Comment by Hotsuyuki** [comment]
Mine also works fine now! Thanks a lot @sjchmiela ! (Deleted the app once and reinstalled from Google play)

**Comment by sohimazdai** [comment]
It works for me: i deleted my expo app from AVD and reinstalled it with google playmarket. Now i do not see the error. Thanks guys!

**Comment by farhanaslam5151** [comment]
![60234247_431319104093915_5167438375459749888_n](https://user-images.githubusercontent.com/38375932/57647393-47d66a80-75dc-11e9-8721-b7529fd55ab3.jpg)

app is working fine on expo client when make expo:build after installing into android giving this error

**Comment by akinolu52** [comment]
but i think this is bad for users.. how will i develop an app and expect users to delete and install again for like 10 times before they can have access to my application. this is a shame!

**Comment by sjchmiela** [comment]
Most likely it's some kind of your app's code error. To ensure your code will work properly in production mode too, run `expo start` with `--no-dev --minify` flags. The bundling will take longer and development mode will be off, but you will be able to check if there are any problems before pushing to the users.

**Comment by cristhiamCMD** [comment]
The same thing happened to me on Android. I went to the EXPO application settings to clear the cache and data of the installed application from PlayStore, and then I started without problems.
I hope it is useful.

**Comment by 509dave16** [comment]
@sjchmiela
### Summary
My co-workers and I have only just now seen this fatal error for the first time when the Expo Client is booting . Currently only one of us is experiencing the issue on Android:
**Device:** Pixel 2 XL
**Android OS:** 9
**Expo Client:** 2.11.2

![Screenshot_20190611-150757](https://user-images.githubusercontent.com/2925882/59310090-252a8680-8c5a-11e9-8e00-3f3113f36640.png)

### Reproduction Steps
1. Close Expo Client
2. Clear Android data and cache for Expo Client
3. Open Expo Client
4. Login to customer's Expo account
5. Close Expo Client
6. Re-Open Expo Client

Basically as long as we're not logged in to the customer's Expo account, we can open and close the Expo Client as many times as we want. But once logged in, if you close the Expo Client and then re-open the fatal error occurs.

**Comment by ide** [comment]
@509dave16 The root cause for the recent issue was fixed earlier today. If you re-open the client, you should no longer encounter this error. Alternatively, you can try re-installing the client although you shouldn't have to.
